packages/astro/src/types/public/config.ts

```diff
@@ -2246,83 +2247,173 @@ export interface ViteUserConfig extends OriginalViteUserConfig {
 					 */
 					styleDirective?: {
 						/**
 						 * @name experimental.csp.styleDirective.hashes
 						 * @type {CspHash[]}
 						 * @default `[]`
-						 * @version 5.5.x
+						 * @version 5.9.0
 						 * @description
 						 *
-						 * A list of style hashes to include in all pages. These hashes are added to the `style-src` directive.
+						 * A list of additional hashes added to the `style-src` directive.
 						 *
+						 * If you have external styles that aren't generated by Astro, this configuration option allows you to provide additional hashes to be rendered. 
+						 *
+						 * You must provide hashes that start with `sha384-`, `sha512-` or `sha256-`. Other values will cause a validation error. These hashes are added to all pages.
+						 *			   
+						 * ```js
+						 * import { defineConfig } from 'astro/config';
+						 *
+						 * export default defineConfig({
+						 *   experimental: {
+						 *     csp: {
+						 *       styleDirective: {
+						 *         hashes: [
+						 *           "sha384-styleHash",
+						 *           "sha512-styleHash",
+						 *           "sha256-styleHash"
+						 *         ]
+						 *       }
+						 *     }
+						 *   }
+						 * });
+						 * ```
 						 */
 						hashes?: CspHash[];
 
 						/**
 						 * @name experimental.csp.styleDirective.resources
 						 * @type {string[]}
 						 * @default `[]`
-						 * @version 5.5.x
+						 * @version 5.9.0
 						 * @description
 						 *
-						 * A list of resources applied to the `style-src` directive.
+						 * A list of resources applied to the `style-src` directive. These resources are added to all pages and will override Astro's defaults.
+						 *
+						 * ```js
+						 * import { defineConfig } from 'astro/config';
 						 *
+						 * export default defineConfig({
+						 *   experimental: {
+						 *     csp: {
+						 *       styleDirective: {
+						 *         resources: [
+						 *           "self",
+						 *           "https://styles.cdn.example.com"
+						 *         ]
+						 *       }
+						 *     }
+						 *   }
+						 * });
+						 * ```
 						 */
 						resources?: string[];
 					};
 
 					/**
-					 * @name experimental.csp.styleDirective
-					 * @type {CspHash[]}
+					 * @name experimental.csp.scriptDirective
+					 * @type {{ hashes?: CspHash[], resources?: string[], strictDynamic?: boolean }}
 					 * @default `undefined`
-					 * @version 5.5.x
+					 * @version 5.9.0
 					 * @description
 					 *
-					 * Allows controlling the `style-src` directive.
+					 * A configuration object that allows you to override the default sources for the `script-src` directive
+					 * with the `resources` property, or to provide additional `hashes` to be rendered.
 					 *
+					 * These properties are added to all pages and completely override Astro's default resources, not add to them.
+					 * Therefore, you must explicitly specify any default values that you want to be included.				 *
```

Comment:
```suggestion
					 * Therefore, you must explicitly specify any default values that you want to be included.
					  *
```
Maybe a missed return here?